scripts/ci/pre_commit/pre_commit_check_setup_installation.rst.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Checks if all the libraries in setup.py are listed in installation.rst file
+"""
+
+import os
+import re
+import sys
+from os.path import dirname
+
+AIRFLOW_SOURCES_DIR = os.path.join(dirname(__file__), os.pardir, os.pardir, os.pardir)
+SETUP_PY_FILE = 'setup.py'
+DOCS_FILE = 'installation.rst'
+PY_IDENTIFIER = '[a-zA-Z_][a-zA-Z0-9_\.]*'
+
+def get_file_content(*file_path: str) -> str:
+    file_path = os.path.join(AIRFLOW_SOURCES_DIR, *file_path)
+    with open(file_path) as file_to_read:
+        return file_to_read.read()
+
+def get_extras_from_setup() -> {str: bool}:
+    """
+    Returns an array EXTRAS_REQUIREMENTS with aliases from setup.py file in format:
+    {'package name': ['alias1', 'alias2], ...}
+    """
+    setup_content = get_file_content(SETUP_PY_FILE)
+    
+    extras_section_regex = re.compile(
```

Comment:
I was experimenting with `from setup import EXTRAS_REQUIREMENTS`, but this was I wasn't able to detect aliases.